Allow any combination of input format and output format.

Sjlver opened the pull request:
There's no technical reason why this shouldn't be allowed.

Closes https://github.com/BlockchainCommons/seedtool-cli/issues/63

Sjlver committed:
Allow any combination of input format and output format.

There's no technical reason why this shouldn't be allowed.

Closes https://github.com/BlockchainCommons/seedtool-cli/issues/63

Sjlver committed:
Remove bip39-to-sskr example script.

It's not needed anymore, since seedtool-cli now allows any combination
of input and output format.

wolfmcnally reviewed (changes requested):
You need to add many more tests to prove that any input format to any output format also work.

Sjlver committed:
Add a test for all meaningful input/output format combinations.

Sjlver commented:
> You need to add many more tests to prove that any input format to any output format also work.

That's a bit of a high standard, considering that there were no tests before ;-)

Anyway, I've added a testcase that tests all meaningful combinations of input/output formats. Let me know what you think.

The test might actually have discovered a bug; shouldn't the output of these two commands be the same? Note that the checksum words are different.

```shell
$ src/seedtool --deterministic TEST | src/seedtool --deterministic TEST --out sskr
tuna acid epic gyro many meow able able able next edge lamb liar city girl draw visa roof logo jolt city waxy jury trip dark safe arch flew what
$ src/seedtool --deterministic TEST | src/seedtool --deterministic TEST --in hex --out sskr
tuna acid epic gyro edge next able able able next edge lamb liar city girl draw visa roof logo jolt city waxy jury trip dark loud song main atom
```

ChristopherA commented:
> > You need to add many more tests to prove that any input format to any output format also work.
> 
> That's a bit of a high standard, considering that there were no tests before ;-)
> 
> Anyway, I've added a testcase that tests all meaningful combinations of input/output formats. Let me know what you think.

I believe all the tests are here: https://github.com/BlockchainCommons/seedtool-cli/blob/master/src/test.sh

Sjlver committed:
Reduce duplicated code in FormatSSKR::process_output.

Sjlver committed:
Fix testAllInputOutputCombinations.

It was missing a `--in hex`, causing it to use random input for SSKR.
This happened to *almost* work because the deterministic seed meant
"random" produced the same result as reading from stdin.

Sjlver commented:
I have fixed the test now.
What I had thought to be a bug was actually not a bug, just an artifact of using the same deterministic seed in two invocations of seedtool-cli.

@wolfmcnally what do you think? Would this be good enough or do you need more tests than that?

Sjlver commented:
I'm planning to close this PR. I still think it makes sense, but since there is no progress, I'm abandoning it.

It you ever feel like supporting this and continuing the work on seedtool.info, let me know.
